### PR TITLE
[cleanup] silence useless cast warnings

### PR DESCRIPTION
**Pull request by Rechi**
## Description
remove cast which do nothing as the (return) type is already the expected one

## Motivation and Context
I wanted to apply `google-readability-casting` clang tidy check, but it removed some casts. So I first identified the casts which are really useless.

## How Has This Been Tested?
compiled for windows and linux

## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cosmetic change (non-breaking change that doesn't touch code)
- [ ] None of the above (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -1845,7 +1845,7 @@ bool CVideoPlayer::GetCachingTimes(double& level, double& delay, double& offset)
 
   delay  = 0.0;
   level  = 0.0;
-  offset = (double)(cached + queued) / length;
+  offset = (cached + queued) / length;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: not sure about this. cached is of type uint64_t. means all operation will be done to this type?

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp [hunk 2]
```diff
@@ -581,8 +581,8 @@ bool CVaapi2Texture::Map(CVaapiRenderPicture* pic)
 
     CEGLAttributes<8> attribs; // 6 static + 2 modifiers
     attribs.Add({{EGL_LINUX_DRM_FOURCC_EXT, static_cast<EGLint>(layer.drm_format)},
-      {EGL_WIDTH, static_cast<EGLint>(width)},
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp]: this can break on systems where EGLint != int
@pkerling ?

Comment (Rechi) [reply xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp]: width is already of type EGLint
https://github.com/xbmc/xbmc/blob/b5069cec5e44bf554be74700adfd596f9403f451/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp#L553-L554

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/cores/VideoPlayer/VideoPlayer.cpp]: uint64_t + double = double

**Review by FernetMenta — APPROVED** [review]

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp (hunk 2 above)
Comment (pkerling) [reply xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp]: :+1:

probably oversight or later modification

**Pull request merged**